### PR TITLE
fix OwnerReference Kind and APIVersion from pr 115

### PR DESCRIPTION
**Pull request by jeffvance**
#115 had a bug where the OwnerReference Kind and APIVersion fields were empty which caused Secret and CM validation to fail.

### CODE REVIEW

File: pkg/provisioner/helpers.go [hunk 1]
```diff
@@ -68,12 +68,10 @@ func claimForKey(key string, c versioned.Interface) (obc *v1alpha1.ObjectBucketC
...
 	obc, err = c.ObjectbucketV1alpha1().ObjectBucketClaims(ns).Get(name, metav1.GetOptions{})
 	if err != nil {
-		if errors.IsNotFound(err) {
-			return nil, err
-		}
-		return nil, fmt.Errorf("error getting claim: %v", err)
+		return nil, err
 	}
-	return obc.DeepCopy(), nil
+	// WHY DeepCopy?? return obc.DeepCopy(), nil
```
Comment (jeffvance) [inline pkg/provisioner/helpers.go]: @copejon

File: pkg/apis/objectbucket.io/v1alpha1/register.go [hunk 2]
```diff
@@ -14,6 +14,10 @@ import (
 // SchemeGroupVersion is group version used to register these objects
 var SchemeGroupVersion = schema.GroupVersion{Group: objectbucketio.GroupName, Version: "v1alpha1"}
 
+var OBCgroupVersion = SchemeGroupVersion.Group + "/" + SchemeGroupVersion.Version
```
Comment (copejon) [inline pkg/apis/objectbucket.io/v1alpha1/register.go]: redundant, can be gotten by

`SchemeGroupVersion.String()`

File: pkg/apis/objectbucket.io/v1alpha1/register.go [hunk 3]
```diff
@@ -14,6 +14,10 @@ import (
...
+
+var OBCgvk = schema.GroupVersionKind{Group: SchemeGroupVersion.Group, Version: SchemeGroupVersion.Version, Kind: "ObjectBucketClaim"}
```
Comment (copejon) [inline pkg/apis/objectbucket.io/v1alpha1/register.go]: per bjeans convo,

drop this line.  add

```
const Version = "v1alpha1"

func GroupKindVersion(kind string) schema.GroupVersionKind {
	return SchemeGroupVersion.WithKind(kind)
}
```

Add to objectbucket_types.go

```
const ObjectBucketKind = "ObjectBucket"

func ObjectBucketGVK() schema.GroupVersionKind {
	return GroupKindVersion(ObjectBucketKind)
}
```

add to objectbucketclaim_types.go

```
const ObjectBucketClaimKind = "ObjectBucketClaim"

func ObjectBucketClaimGVK() schema.GroupVersionKind{
	return GroupKindVersion(ObjectBucketClaimKind)
}
```

File: pkg/apis/objectbucket.io/v1alpha1/objectbucket_types.go [hunk 4]
```diff
@@ -109,6 +110,12 @@ type ObjectBucketStatus struct {
 	Conditions corev1.ConditionStatus  `json:"conditions"`
 }
 
+const ObjectBucketKind = "ObjectBucket"
```
Comment (copejon) [inline pkg/apis/objectbucket.io/v1alpha1/objectbucket_types.go]: nit, but this is kind of awkwardly placed between ObjectBucket components.  can you move it to the top or bottom of the file?

File: pkg/apis/objectbucket.io/v1alpha1/objectbucketclaim_types.go [hunk 5]
```diff
@@ -86,6 +87,12 @@ type ObjectBucketClaim struct {
 	Status ObjectBucketClaimStatus `json:"status,omitempty"`
 }
 
+const ObjectBucketClaimKind = "ObjectBucketClaim"
```
Comment (copejon) [inline pkg/apis/objectbucket.io/v1alpha1/objectbucketclaim_types.go]: ditto placement

File: pkg/provisioner/util_test.go [hunk 6]
```diff
@@ -493,7 +493,7 @@ func TestNewBucketConfigMap(t *testing.T) {
 // 					StorageClassName: className,
 // 					ReclaimPolicy:    &deletePolicy,
 // 					ClaimRef: &corev1.ObjectReference{
-// 						Kind:            "ObjectBucketClaim",
+// 						Kind:            v1alpha1.OBCKind,
```
Comment (copejon) [inline pkg/provisioner/util_test.go]: undefined variable

**Pull request merged**